Lib/test/test_uuid.py

```diff
@@ -657,6 +658,11 @@ def testIssue8621(self):
 
             self.assertNotEqual(parent_value, child_value)
 
+    # bpo-35701: check that weak referencing to a UUID object can be created
+    def test_uuid_weakref(self):
+        strong = self.uuid.uuid4()
+        weak = weakref.ref(strong)
+        assert strong is weak()
```

Comment:
Please replace assert with self.assertIs().

Reply:
Of course, may I ask why?

Reply:
"assert" is removed when running python -O. Moreover, assertIs() provides better error messages on failure (render the two variables).